addons/mail_group/models/mail_group.py

```diff
@@ -0,0 +1,570 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import base64
+import json
+import logging
+
+from ast import literal_eval
+from datetime import datetime
+from dateutil import relativedelta
+from werkzeug import urls
+from odoo.addons.http_routing.models.ir_http import slug
+
+from odoo import _, api, fields, models, modules, tools
+from odoo.exceptions import ValidationError, UserError
+from odoo.osv import expression
+from odoo.tools import email_normalize, hmac, generate_tracking_message_id
+
+_logger = logging.getLogger(__name__)
+
+GROUP_SEND_BATCH_SIZE = 500
+
+
+class MailGroup(models.Model):
+    """This model represents a mailing list.
+
+    Users send emails to an alias to create new group messages or reply to existing
+    group messages. Moderation can be activated on groups. In that case email have to
+    be validated or rejected.
+    """
+    _name = 'mail.group'
+    _description = 'Mailing List'
+    # TDE CHECK: use blaclist mixin
+    _inherit = ['mail.alias.mixin']
+
+    @api.model
+    def default_get(self, fields):
+        res = super(MailGroup, self).default_get(fields)
+        if not res.get('alias_contact') and (not fields or 'alias_contact' in fields):
+            res['alias_contact'] = 'everyone' if res.get('access_mode') == 'public' else 'followers'
+        return res
+
+    def _get_default_image(self):
+        image_path = modules.get_module_resource('mail', 'static/src/img', 'groupdefault.png')
+        return base64.b64encode(open(image_path, 'rb').read())
```

Comment:
`file_open` pretty much does the lookup for you internally.

Reply:
Noted for "cleaning" task.

Reply:
Ah seems Chs is reveilled donc plein de modifs donc tankafer...